### PR TITLE
ProxySChunk

### PR DESCRIPTION
**Pull request by martaiborra**

### CODE REVIEW

File: blosc2/proxyschunk.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+# LICENSE file in the root directory of this source tree)
+#######################################################################
+import blosc2
+from abc import ABC, abstractmethod
+
+
+class ProxySource(ABC):
+    """
+    Base interface for all supported sources in :ref:`ProxySChunk`.
+    In case the source is multidimensional, the attributes `shape`, `chunks`,
```
Comment (FrancescAlted) [inline blosc2/proxyschunk.py:15]: PEP 8 mandates a blank line between brief description (one line) and the extended descrition.

File: blosc2/proxyschunk.py [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+    `blocks` and `dtype` are also required when creating the :ref:`ProxySChunk`.
```
Comment (FrancescAlted) [inline blosc2/proxyschunk.py:16]: For clarity, I'd add a blank line after here.

File: blosc2/proxyschunk.py [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+        Returns
+        -------
+        out: bytes object
+            The compressed chunk.
+        """
+        pass
+
+
+class ProxySChunk:
+    """Class that implements a proxy (with cache support) of a Python-Blosc2 container.
```
Comment (FrancescAlted) [inline blosc2/proxyschunk.py:41]: Suggestion "Python-Blosc2 container" -> "Python-Blosc2 super-chunk"

File: doc/reference/proxyschunk.rst [hunk 4]
```diff
@@ -0,0 +1,32 @@
+.. _ProxySChunk:
+
+ProxySChunk
+===========
+
+Class that implements a proxy (with cache support) of a Python-Blosc2 container.
```
Comment (FrancescAlted) [inline doc/reference/proxyschunk.rst:6]: Ditto

File: doc/reference/proxyschunk.rst [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+This can be used to cache chunks of
+a regular data container which follows the :ref:`ProxySource` interface in an urlpath.
```
Comment (FrancescAlted) [inline doc/reference/proxyschunk.rst:9]: "regular data container" -> "regular super-chunk"

File: doc/reference/proxyschunk.rst [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    afetch
+
+Attributes
+----------
+
+.. autosummary::
+    :toctree: autofiles/proxyschunk
+
+    shape
+    dtype
```
Comment (FrancescAlted) [inline doc/reference/proxyschunk.rst:32]: Is `vlmeta` missing here?

File: doc/reference/proxysource.rst [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+===========
+
+Base interface for all supported sources in :ref:`ProxySChunk`.
+
+In case the source is multidimensional, the attributes `shape`, `chunks`,
+`blocks` and `dtype` are also required when creating the :ref:`ProxySChunk`.
+In case the source is unidimensional, the attributes `chunksize`, `typesize`
+and `nbytes` are required as well when creating the :ref:`ProxySChunk`.
+These attributes do not need to be available when opening an already
+existing :ref:`ProxySChunk`.
```
Comment (FrancescAlted) [inline doc/reference/proxysource.rst:13]: This docstring is duplicated from `ProxySChunk` description.  Could sphinx be instructed to insert the description here?

File: tests/test_schunk_proxy.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
+#######################################################################
```
Comment (FrancescAlted) [inline tests/test_schunk_proxy.py:1]: I think a more consistent name for this module would be `test_proxy_schunk.py`

File: tests/ndarray/test_c2array_proxy.py [hunk 9]
```diff
@@ -0,0 +1,95 @@
+#######################################################################
```
Comment (FrancescAlted) [inline tests/ndarray/test_c2array_proxy.py:1]: For consistency, I'd prefer this module to be called `test_proxy_c2array.py`

**Review by FrancescAlted — COMMENTED** [review]
See my comments.  Other than that, this looks good to me.

File: doc/reference/proxyschunk.rst (hunk 6 above)
Comment (martaiborra) [reply doc/reference/proxyschunk.rst]: Yes, fixed in 42c27b8bc76bc1c4ed98a708df63fec788dd1b0a.

File: tests/test_schunk_proxy.py (hunk 8 above)
Comment (martaiborra) [reply tests/test_schunk_proxy.py]: Fixed in 42c27b8bc76bc1c4ed98a708df63fec788dd1b0a.

File: tests/ndarray/test_c2array_proxy.py (hunk 9 above)
Comment (martaiborra) [reply tests/ndarray/test_c2array_proxy.py]: Fixed in 42c27b8bc76bc1c4ed98a708df63fec788dd1b0a.

**Review by FrancescAlted — COMMENTED** [review]
Some comments (although I am not sure we should be too explicit about the object returned when source is not SChunk, NDArray or CArray); I personally would remove it, as it may create more confusion than clarity.

File: doc/reference/proxysource.rst (hunk 7 above)
Comment (martaiborra) [reply doc/reference/proxysource.rst]: It is possible but the structure and design of attributes, methods, etc, will be different from the other classes.

**Review by FrancescAlted — APPROVED** [review]
LGTM

**Pull request merged**